presto-spi/src/main/java/com/facebook/presto/spi/block/Block.java

```diff
@@ -295,4 +295,12 @@ default Block getLoadedBlock()
     {
         return this;
     }
+
+//    int[] getSizesInBytes();
+//
+//    int[] getSizesInBytes(int[] offsets, int offset, int length);
+
+    void appendPositionSizesInBytes(int[] sizesInBytes);
```

Comment:
What's the semantic of these APIs ? :)

Reply:
@wenleix These two methods populated the row sizes in the passed in int[] sizesInBytes for each row. We pass in this int[] to avoid memory allocation as much as possible.